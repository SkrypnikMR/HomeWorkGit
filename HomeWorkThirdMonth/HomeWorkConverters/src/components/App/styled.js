import styled from 'styled-components';

export const StAppDiv = styled.div`
background-color: #27332F;
color: white;
min-height: 100vh;
width: 100%;
display: flex;
justify-content: center;
align-items: center;
flex-direction: column;
`;
