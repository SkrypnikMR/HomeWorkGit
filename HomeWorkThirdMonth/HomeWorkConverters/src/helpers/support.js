export const support = {
    lsGet: name => localStorage.getItem(name),
    lsSet: (name, item) => {
        if (typeof item === 'string') return localStorage.setItem(name, item);
        localStorage.setItem(name, JSON.stringify(item));
    },
    convert: (num, primary, secondary) => {
        const lengthSystem = {
            meters: {
                meters: 1,
                miles: 1609.344,
                yards: 1.093613,
                feet: 3.28084,
                versts: 1066.8,
            },
            miles: {
                meters: 1609.344,
                miles: 1,
                yards: 1760,
                feet: 0.003281,
                versts: 1.50857,
            },
            yards: {
                meters: 0.9144,
                miles: 0.000568182,
                yards: 1,
                feet: 3,
                versts: 0.000857143,
            },
            feet: {
                meters: 0.3048,
                miles: 0.000189394,
                yards: 0.333333,
                feet: 1,
                versts: 0.000285714,
            },
            versts: {
                meters: 1066.8,
                miles: 0.662879,
                yards: 1166.67,
                feet: 3500,
                versts: 1,
            },
        };
        return (+num * lengthSystem[primary][secondary]).toString();
    },
};
