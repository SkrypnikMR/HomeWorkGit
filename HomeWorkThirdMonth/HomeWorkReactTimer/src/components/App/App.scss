.app-dark, .app-light{
    font-family: 'Play', sans-serif;
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    flex-direction: column;
}
.app-light{
    background-color: rgb(229, 208, 237);
}
.app-dark{
    background-color: rgb(91, 88, 92);
}
