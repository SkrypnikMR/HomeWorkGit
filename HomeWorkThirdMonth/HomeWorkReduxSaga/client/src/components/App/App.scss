body {
    font-family: 'Play', sans-serif;
    margin: 0 auto;
   	background-color: #1B264F;
}
.app {
    display: flex;
    flex-direction: column;
    height: 100vh; 
}