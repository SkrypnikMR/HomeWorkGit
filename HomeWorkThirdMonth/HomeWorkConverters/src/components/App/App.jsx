import React from 'react';
import { StAppDiv } from './styled';
import Header from '../Header';

const App = () => {
    return (
        <StAppDiv>
            <Header />
        </StAppDiv>
    );
};

export default App;
