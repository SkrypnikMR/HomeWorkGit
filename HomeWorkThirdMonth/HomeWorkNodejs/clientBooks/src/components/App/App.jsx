import React, { Component } from 'react';
import Header from '../Header';
import Content from '../Content';
import './App.scss';

import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
class App extends Component {
    constructor() {
        super();
    }
    render() {
        return (
            <div className="app">
                <Router>
                    <Header />
                    <Switch>
                        <Route exact path='/'><Content/></Route>
                    </Switch>
                </Router>
            </div>
        )
    }
}

export default App;

