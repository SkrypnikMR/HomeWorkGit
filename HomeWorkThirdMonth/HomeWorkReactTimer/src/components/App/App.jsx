import React, { Component } from 'react';
import Header from '../Header';
import Main from '../Main';
import Footer from '../Footer';
import { ThemeContext, themes } from '../../context/themeContext';
import './App.scss';
import { icon } from '../../helper/iconCreator';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = { theme: themes.light, bulb: icon(themes.light) };
    }

    toggleTheme() {
        const { theme } = this.state;
        if (theme === themes.light) return this.setState({ theme: themes.dark, bulb: icon(themes.dark) });
        this.setState({ theme: themes.light, bulb: icon(themes.light) });
    }

    render() {
        const { theme, bulb } = this.state;
        return (
            <ThemeContext.Provider value={{ toggleTheme: this.toggleTheme.bind(this), bulb }}>
                <div className={`app-${theme}`}>
                    <Header />
                    <Main />
                    <Footer />
                </div>
            </ThemeContext.Provider >
        );
    }
}

export default App;
