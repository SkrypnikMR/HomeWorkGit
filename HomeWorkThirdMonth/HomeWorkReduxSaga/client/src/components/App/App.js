import React from 'react';
import Header from '../Header';
import Movies from '../Movies';
import './App.scss';
import Footer from '../Footer';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';

const App = () => (
    <div className='app'>
        <Header />
        <Movies />
        <Footer />
        <NotificationContainer />
    </div>
);

export default App;
