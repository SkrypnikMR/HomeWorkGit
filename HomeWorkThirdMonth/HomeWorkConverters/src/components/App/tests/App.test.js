import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { shallowSmart } from '../../../../__tests__/helpers/testHelper';
import App from '../App';

jest.mock('react-modal');

const mockStore = configureStore();
const store = mockStore({
    isLoading: false,
    converterType: 'lengths',
    from: '',
    to: '',
    quantity: '',
    tablo: '',
    currency: {},
    notes: [],
});

describe('Button', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<App />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render Header', () => {
        const component = shallow(<App />);
        expect(component.find('Header')).toHaveLength(1);
    });
    it('should render Converter', () => {
        const component = shallow(<App />);
        expect(component.find('Converter')).toHaveLength(1);
    });
});
