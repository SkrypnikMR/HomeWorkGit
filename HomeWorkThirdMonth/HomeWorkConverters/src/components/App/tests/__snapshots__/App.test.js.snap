// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot 1`] = `"<div class=\\"styled__StAppDiv-sc-7r44pg-0 gzsdRD\\"><div class=\\"styled__StHeader-zzrvsn-0 fMlHVa\\"><h1>Converter</h1><div class=\\"styled__StControl-sc-13jhp11-0 hMnvOa\\"><h1>Change converter to</h1><button id=\\"lengths\\" name=\\"converter_Lengths\\" type=\\"button\\" disabled=\\"\\" height=\\"50px\\" width=\\"100px\\" class=\\"styled__StButton-sc-1a6hck8-0 kBWHlo\\">Length</button><h1>Or</h1><button id=\\"curency\\" name=\\"converter_Currency\\" type=\\"button\\" height=\\"50px\\" width=\\"100px\\" class=\\"styled__StButton-sc-1a6hck8-0 kBWHlo\\">Currency</button><button id=\\"modal_btn\\" name=\\"modal_btn\\" type=\\"button\\" height=\\"50px\\" width=\\"100px\\" class=\\"styled__StButton-sc-1a6hck8-0 kBWHlo\\">Show All Results</button></div></div><div class=\\"styled__StConverter-ljip47-0 iQrdBt\\"><div class=\\"styled__StFormDiv-sc-1xa63f-0 ejFTTT\\"><div class=\\"styled__StTablo-sc-1dxfj2m-0 fuzSsk\\"></div><div class=\\"styled__StConvertControl-sc-19u337l-0 kJUHeb\\"><div class=\\"styled__StDropDownsControl-sc-19u337l-1 eyBSTn\\"><select width=\\"60%\\" class=\\"styled__StSelect-sc-1k7m7de-0 cJrwdK\\"><option id=\\"trueMETERS\\" value=\\"meters\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">METERS</option><option id=\\"trueVERSTS\\" value=\\"versts\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">VERSTS</option><option id=\\"trueMILES\\" value=\\"miles\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">MILES</option><option id=\\"trueFOOTS\\" value=\\"foots\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">FOOTS</option><option id=\\"trueYARDS\\" value=\\"yards\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">YARDS</option>;</select><p>TO</p><select width=\\"60%\\" class=\\"styled__StSelect-sc-1k7m7de-0 cJrwdK\\"><option id=\\"trueMETERS\\" value=\\"meters\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">METERS</option><option id=\\"trueVERSTS\\" value=\\"versts\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">VERSTS</option><option id=\\"trueMILES\\" value=\\"miles\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">MILES</option><option id=\\"trueFOOTS\\" value=\\"foots\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">FOOTS</option><option id=\\"trueYARDS\\" value=\\"yards\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">YARDS</option>;</select></div><div width=\\"90%\\" class=\\"styled__StInputContainer-w3ykst-3 bwsXTy\\"><input type=\\"text\\" id=\\"convertControlPanelInput\\" value=\\"\\" placeholder=\\"Enter your param\\" class=\\"styled__StInput-w3ykst-1 xGBPx\\"/></div><span class=\\"styled__StSpan-sc-19u337l-2 iDQTBr\\"></span><button id=\\"controllButton\\" type=\\"button\\" class=\\"styled__StButton-sc-1a6hck8-0 cqrnfx\\">Convert</button></div></div></div></div>"`;
